docs/ref/contrib/gis/gdal.txt

```diff
@@ -1101,35 +1104,107 @@ one or more layers of data named bands.  Each band, represented by a
 image is represented as three bands: one for red, one for green, and one for
 blue.
 
-.. class:: GDALRaster(ds_input)
+.. note::
 
-    The constructor for ``GDALRaster`` accepts a single parameter: the path of
-    the file you want to read.
+    For raster data there is no difference between a raster instance and its
+    data source. Unlike for the Geometry objects, :class:`GDALRaster` objects are
+    always a data source. Temporary rasters can be instantiated in memory
+    using the corresponding driver, but they will be of the same class as file-based
+    raster sources.
+
+.. class:: GDALRaster(ds_input, write=False)
+
+    .. versionchanged:: 1.9
```

Comment:
I generally prefer putting the "changed" annotations at the bottom of the section (above `.. attribute:: name` in this case)